### PR TITLE
Terraform dashboard

### LINKED ISSUE #90: Terraform Dashboard
### Task

- [x] Dockerfile for dashboard
- [x] Upload image to ECR
- [x] Create an ECS fargate service to run the dashboard with terraform
- [x] Update README 

### User Story
As a developer, I want to automate the process of deploying a dashboard to an ECS service on AWS so that I can maintain and repeat this process if need be.

### PR DESCRIPTION
**Pull request by FarihaChoudhury**
Terraform ECS for running dashboard.

closes #90

### CODE REVIEW

File: dashboard/dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+
+
+# AWS ECR PUSH COMMANDS:
+    # aws ecr get-login-password --region eu-west-2 | docker login --username AWS --password-stdin 129033205317.dkr.ecr.eu-west-2.amazonaws.com
+    # docker build --platform "linux/amd64" -t c11-railway-tracker-dashboard-erc .
+    # docker tag c11-railway-tracker-dashboard-erc:latest 129033205317.dkr.ecr.eu-west-2.amazonaws.com/c11-railway-tracker-dashboard-erc:latest
+    # docker push 129033205317.dkr.ecr.eu-west-2.amazonaws.com/c11-railway-tracker-dashboard-erc:latest
+
```
Comment (Zhi-704) [inline dashboard/dockerfile]: I cannot believe this travesty

Comment (Zhi-704) [reply dashboard/dockerfile]: Put this in the bin

File: dashboard/dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+FROM python:latest
```
Comment (Zhi-704) [inline dashboard/dockerfile]: It's good to specify the python version

Comment (Zhi-704) [reply dashboard/dockerfile]: Maybe python:3.12-slim?

Comment (Zhi-704) [reply dashboard/dockerfile]: You cannot silence me

File: dashboard/dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+COPY requirements.txt .
+RUN pip3 install -r requirements.txt
+
+EXPOSE 8501 
+
+RUN mkdir ./pages
+COPY ./pages ./pages
+
+RUN mkdir ./streamlit
+COPY ./.streamlit ./.streamlit
```
Comment (A-Faris) [inline dashboard/dockerfile:14]: This could probably be improved

File: dashboard/dockerfile (hunk 1 above)
Comment (FarihaChoudhury) [reply dashboard/dockerfile]: im so sorry pls don't hurt me

File: dashboard/dockerfile (hunk 2 above)
Comment (FarihaChoudhury) [reply dashboard/dockerfile]: shush

File: dashboard/dockerfile (hunk 3 above)
Comment (FarihaChoudhury) [reply dashboard/dockerfile:14]: that's true

**Review by Zhi-704 — DISMISSED** [review]
Dockerfile should be changed

**Review by A-Faris — DISMISSED** [review]
Everything looks good (except the dockerfile)

**Review by Zhi-704 — APPROVED** [review]

**Pull request merged**